### PR TITLE
chore: add signer check nullify update address mt rollovers

### PR DESCRIPTION
**Pull request by ananas-block**
Changes:
- regenerated accounts
- add mint to supply check (that the correct amount of tokens was actually minted)
- add signer check for program to be registered pubkey in register program
- add signer or is registered program check for nullify, update address merkle tree, rollover instructions
- add failing tests for signer with invalid group for all account compression instructions

### CODE REVIEW

File: programs/compressed-token/src/process_mint.rs [hunk 1]
```diff
@@ -330,6 +330,20 @@ pub fn mint_spl_to_pool_pda<'info>(
     let cpi_ctx = CpiContext::new(ctx.accounts.token_program.to_account_info(), cpi_accounts);
 
     anchor_spl::token::mint_to(cpi_ctx, mint_amount)?;
+    let post_token_balance = TokenAccount::try_deserialize(
+        &mut &ctx.accounts.token_pool_pda.to_account_info().data.borrow()[..],
+    )?
+    .amount;
+    // Guard against unexpected behavior of the token program.
```
Comment (SwenSchaeferjohann) [inline programs/compressed-token/src/process_mint.rs:337]:
```suggestion
    // Guard against unexpected behavior of the SPL token program.
```

File: programs/compressed-token/src/process_transfer.rs [hunk 2]
```diff
@@ -202,7 +202,7 @@ pub fn add_token_data_to_input_compressed_accounts(
 // TODO: consider moving this function to helpers
 /// Get static cpi signer seeds
 pub fn get_cpi_signer_seeds() -> [&'static [u8]; 2] {
-    let bump: &[u8; 1] = &[255];
+    let bump: &[u8; 1] = &[254];
     let seeds: [&'static [u8]; 2] = [b"cpi_authority", bump];
     seeds
 }
```
Comment (SwenSchaeferjohann) [inline programs/compressed-token/src/process_transfer.rs:208]: why the change?

Comment (ananas-block) [reply programs/compressed-token/src/process_transfer.rs:208]: different program id different derivation

File: test-programs/account-compression-test/tests/group_authority_tests.rs [hunk 3]
```diff
@@ -160,24 +163,23 @@ async fn test_create_and_update_group() {
...
-        compressed_pda_id
+        system_program_id_keypair.pubkey()
     );
     assert_eq!(
         registerd_program_account.group_authority_pda,
         group_accounts.0
     );
     // add new program to group with invalid authority
-    let other_program_id = Pubkey::new_unique();
+    let other_program_id = Keypair::new();
```
Comment (SwenSchaeferjohann) [inline test-programs/account-compression-test/tests/group_authority_tests.rs:173]:
```suggestion
    let other_program_id = Keypair::new().pubkey();
```

File: test-programs/account-compression-test/tests/group_authority_tests.rs [hunk 4]
```diff
@@ -160,24 +163,23 @@ async fn test_create_and_update_group() {
...
     let registered_program_pda =
-        Pubkey::find_program_address(&[other_program_id.to_bytes().as_slice()], &ID).0;
+        Pubkey::find_program_address(&[other_program_id.pubkey().to_bytes().as_slice()], &ID).0;
```
Comment (SwenSchaeferjohann) [inline test-programs/account-compression-test/tests/group_authority_tests.rs:175]:
```suggestion
        Pubkey::find_program_address(&[other_program_id.to_bytes().as_slice()], &ID).0;
```

File: test-programs/account-compression-test/tests/group_authority_tests.rs [hunk 5]
```diff
@@ -160,24 +163,23 @@ async fn test_create_and_update_group() {
...
 
-    let register_program_ix = account_compression::instruction::RegisterProgramToGroup {
-        program_id: other_program_id,
-    };
+    let register_program_ix = account_compression::instruction::RegisterProgramToGroup {};
     let instruction = Instruction {
         program_id: ID,
         accounts: vec![
             AccountMeta::new(context.get_payer().pubkey(), true),
+            AccountMeta::new(other_program_id.pubkey(), true),
```
Comment (SwenSchaeferjohann) [inline test-programs/account-compression-test/tests/group_authority_tests.rs:182]:
```suggestion
            AccountMeta::new(other_program_id, true),
```

File: test-programs/account-compression-test/tests/group_authority_tests.rs [hunk 6]
```diff
@@ -189,7 +191,7 @@ async fn test_create_and_update_group() {
     let transaction = Transaction::new_signed_with_payer(
         &[instruction],
         Some(&context.get_payer().pubkey()),
-        &vec![&context.get_payer()],
+        &vec![&context.get_payer(), &other_program_id],
```
Comment (SwenSchaeferjohann) [inline test-programs/account-compression-test/tests/group_authority_tests.rs:194]: aesthetics: program_keypair ..

File: test-programs/registry-test/tests/tests.rs [hunk 7]
```diff
@@ -10,7 +10,7 @@ use light_test_utils::test_env::{
 async fn test_e2e() {
     let (mut rpc, env) = setup_test_programs_with_accounts(None).await;
     let random_program_id = Keypair::new();
-    register_program_with_registry_program(&mut rpc, &env, &random_program_id.pubkey())
```
Comment (SwenSchaeferjohann) [inline test-programs/registry-test/tests/tests.rs:13]: technically a program_keypair too

**Review by SwenSchaeferjohann — APPROVED** [review]
